src/DateTime.cc

```diff
@@ -436,6 +436,9 @@ DateTime::DateTime(std::string const& iso8601, Timescale scale) {
 }
 
 double DateTime::get(DateSystem system, Timescale scale) const {
+    if (!isValid()) {
+        throw LSST_EXCEPT(pexEx::RuntimeError, "DateTime not valid");
```

Comment:
I'm not keen on custom exceptions unless they are really needed. In this case `RuntimeError` seems to reflect the fact that the user should not have performed the action on an invalid date. (I deleted an earlier comment that was meant to apply to your suggestion of adding an _assertValid method, which I definitely will do).